### DISCUSSION TITLE
How to append bindings to fill holes?

### DISCUSSION THREAD

**Discussion by werto87** [SOCI/soci#956] [General]
In this example all holes get filled at once:
```cpp
string firstName = "John", lastName = "Smith";
int personId = 7;

sql << "insert into person(id, firstname, lastname) values(:id, :fn, :ln)",
        use(personId), use(firstName), use(lastName);
```
I want to append the "use" step by step.
Something like this:
```cpp
string firstName = "John", lastName = "Smith";
int personId = 7;

MyStatement myStatement{};
myStatement.query= "insert into person(id, firstname, lastname) values(:id, :fn, :ln)";
myStatement.appendUse(personId);
myStatement.appendUse(firstName);
myStatement.appendUse(lastName);
myStatement.execute (true);
```

**Details**
I am writing convince functions which are build on soci and boost::fusion.
For example ["insertStruct"](https://github.com/werto87/confu_soci/blob/caf0dd2c629b45ea50ec43e2a8c7c47c6909770c/confu_soci/convenienceFunctionForSoci.hxx#L128) to add one row into the EasyClass table.
```cpp
   insertStruct (sql, EasyClass{ "222", "someValue" });
```
Creating the sql expresion is done by iterating over the members of EasyClass.
So for "insertStruct (sql, EasyClass{ "222", "someValue" });" the sql is:
```sql
INSERT INTO EasyClass(playerId,points) VALUES('222','someValue')
```
This allows sql injections. So I want to use "use()" instead of copying the values of the struct into VALUES().
Is it possible to create the query with holes and fill them by appending?

**Comment by vadz** [comment]
You do it like this:
```cpp
    soci::statement st(session);
    st.alloc();
    st.prepare("insert into ...");

    st.exchange(use(value1));
    st.exchange(use(value2));
    st.exchange(use(value3));

    st.define_and_bind();
    st.execute();
```

**Comment by werto87** [comment]
I checked your suggestion and it works for me. Thank You!
